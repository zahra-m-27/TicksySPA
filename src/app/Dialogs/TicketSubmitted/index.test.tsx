import ReactDOM from 'react-dom';
import React from 'react';
import {render} from '@testing-library/react';
import renderer from 'react-test-renderer';
import TicketSubmittedDialog from './index';

const sample = <TicketSubmittedDialog identifier="Test Identifier" />;

test('render in ReactDOM', () => {
  const div = document.createElement('div');
  ReactDOM.render(sample, div);
});

it('render title', () => {
  const {getByText} = render(sample);
  getByText(/درخواست شما با موفقیت ارسال شد./i);
});

it('render message', () => {
  const {getByText} = render(sample);
  getByText(/برایتان ایمیل خواهد شد/i);
});

it('render identifier label', () => {
  const {getByText} = render(sample);
  getByText(/شناسه درخواست/i);
});

it('render identifier', () => {
  const {getByText} = render(sample);
  getByText(/Test Identifier/i);
});

it('snapshot', () => {
  const snapshot = renderer.create(sample).toJSON();
  expect(snapshot).toMatchSnapshot();
});
