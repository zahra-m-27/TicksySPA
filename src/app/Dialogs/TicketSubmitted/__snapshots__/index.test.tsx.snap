// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="container"
>
  <img
    alt="success"
    className="icon"
    src="checked.png"
  />
  <p
    className="title"
    dir="auto"
  >
    درخواست شما با موفقیت ارسال شد.
  </p>
  <p
    className="message"
    dir="auto"
  >
    درخواست شما ارسال شد و در کوتاه ترین زمان پاسخ آن به شما ارائه خواهد شد. نتیجه درخواست خود را می توانید از داشبورد ببینید هم چنین نتیجه درخواست برایتان ایمیل خواهد شد
  </p>
  <div
    className="id_container"
  >
    <p
      dir="auto"
    >
      شناسه درخواست:
    </p>
    <p
      className="id"
      dir="auto"
    >
      Test Identifier
    </p>
  </div>
</div>
`;
